src/reducers/exercise.js

```diff
@@ -0,0 +1,65 @@
+import * as types from '../constants/actionTypes';
+import { AsyncStorage, Alert } from 'react-native';
+
+const course = require('../assets/courses/course.json');
+
+// make a "timeline" for exercises
+const exercises = [];
+for (let i = 0; i < course.sections.length; i++) {
+  const section = course.sections[i];
+  for (let j = 0; j < section.chapters.length; j++) {
+    const chapter = section.chapters[j];
+    for (let k = 0; k < chapter.exercises.length; k++) {
+      const exercise = chapter.exercises[k];
+      exercise.complete = false; // later take this info from storage
+      exercises.push(exercise);
+    }
+  }
+}
+
+const initialState = {
+  course,
+  exercises,
+};
+
+// init current exercise from storage
+AsyncStorage
```

Comment:
Maybe we could use [`redux-persist`](https://github.com/rt2zz/redux-persist) for that since this info will be part of the store, anyway?